src/functions/color.js

```diff
@@ -102,7 +102,13 @@ defineFunction({
     handler({parser, breakOnTokenText}, args) {
         const color = assertNodeType(args[0], "color-token").color;
 
-        // If we see a styling function, parse out the implicit body
+        // Set macro \current@color in current namespace to store the current
+        // color, mimicking the behavior of color.sty.
+        // This is currently used just to correctly color a \right
+        // that follows a \color command.
+        parser.gullet.macros.set("\\current@color", color);
```

Comment:
macros defined within a group get popped as we exit the group right?

Reply:
Correct. I think this is implied by the current comment `Set macro \current@color in current namespace` but feel free to suggest a better wording.

FWIW, we're already using this mechanism to redefine the meaning of `\\` within array environments. It's quite useful!